Public/Sdk/Experimental/NodeJs/npm.dsc

```diff
@@ -41,6 +41,39 @@ namespace Npm {
         };
     }
 
+    @@public
+    export function installFromPackageJson(workingDirectory : Directory, nodeModulesPath : Directory) : Directory {
+        const arguments: Argument[] = [
+            Cmd.argument(Artifact.input(Node.npmCli)),
+            Cmd.argument("install")
+        ];
+
+        const result = Node.run({
+            arguments: arguments,
+            workingDirectory: workingDirectory,
+            outputs: [
+                nodeModulesPath

```

Comment:
Specifying this folder here doesn't tell `Node` where to place outputs; instead this only declares where outputs are expected to be.  That means that this pip will simply run `npm install` from the working directory (which in your case will be the `client` folder); when this pip is executed, it will try to produce `client/node_modules` and not the folder specified in `nodeModulesPath`.